src/kagglehub/http_resolver.py

```diff
@@ -199,6 +200,66 @@ def _inner_download_file(file: str) -> None:
         return out_path
 
 
+class NotebookOutputHttpResolver(Resolver[CodeHandle]):
+    def is_supported(self, *_, **__) -> bool:  # noqa: ANN002, ANN003
+        # Downloading files over HTTP is supported in all environments for all handles / paths.
+        return True
+
+    def __call__(self, h: CodeHandle, path: Optional[str] = None, *, force_download: Optional[bool] = False) -> str:
+        api_client = KaggleApiV1Client()
+
+        cached_response = load_from_cache(h, path)
+        if cached_response and not force_download:
+            return cached_response  # Already cached
+        elif cached_response and force_download:
+            delete_from_cache(h, path)
+
+        api_path = f"kernels/{h.owner}/{h.notebook}/output/download"
+        output_root = Path(get_cached_path(h, path))
+
+        # Create the intermediary directories
+        # List the files and decide how to download them:
+        # - <= 25 files: Download files in parallel
+        # > 25 files: Download the archive and uncompress
+        (files, has_more) = self._list_files(api_client, h)
+        if has_more:
+            # TODO(b/379761520): support .tar.gz archived downloads
```

Comment:
maybe misplaced, are .tar.gz files supposed to fail at this point?

Reply:
yeah, this is how `model_download` knows when to download an archive vs individual files:

https://github.com/Kaggle/kagglehub/blob/8b1fae8632f9d381cebb14ec50c56d7ff5fbeb1b/src/kagglehub/http_resolver.py#L168-L172

But we don't have an API endpoint that returns the supported compression formats. See the bug for more context.